Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -211,6 +228,41 @@ std::map<std::string, std::string> FindSXPeaks::validateInputs() {
     validationOutput["XResolution"] = "XResolution must be set to a value greater than 0";
   }
 
+  const int minNSpectraPerPeak = static_cast<int>(getProperty("MinNSpectraPerPeak"));
+  const int maxNSpectraPerPeak = static_cast<int>(getProperty("MaxNSpectraPerPeak"));
+  if (!isEmpty(minNSpectraPerPeak) && !isEmpty(maxNSpectraPerPeak)) {
+    if (maxNSpectraPerPeak < minNSpectraPerPeak) {
+      validationOutput["MaxNSpectraPerPeak"] = "MaxNSpectraPerPeak must be greater than MinNSpectraPerPeak";
+      validationOutput["MinNSpectraPerPeak"] = "MinNSpectraPerPeak must be lower than MaxNSpectraPerPeak";
+    }
+  }
+
+  MatrixWorkspace_const_sptr inputWorkspace = getProperty("InputWorkspace");
+  if (inputWorkspace) {
+    size_t numberOfSpectra = inputWorkspace->getNumberHistograms();
```

Comment:
Here the user could supply a `StartWorkspaceIndex` and `EndWorkspaceIndex` - so the number of spectra considered can be less than the total number of histograms if these are specified

Reply:
Thank you very much for the detailed review! This validation has been added.